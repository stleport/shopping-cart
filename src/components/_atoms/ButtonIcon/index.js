import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';
import Button from '../Button';

const ButtonIcon = ({ 
  iconName, 
  iconColor, 
  buttonStyle, 
  onChangeQuantity
}) => (
  (
    <React.Fragment>
      <Button 
        onClickButton={onChangeQuantity} 
        buttonStyle={buttonStyle}
      >
        <Icon 
          iconName={iconName} 
          iconStyle={`st-Edit-button__icon--${iconColor}`}
        />
      </Button> 
    </React.Fragment>
  )
);

ButtonIcon.propTypes = {
  iconName: PropTypes.string.isRequired,
  iconColor: PropTypes.string.isRequired,
  buttonStyle: PropTypes.string.isRequired,
  onChangeQuantity: PropTypes.func.isRequired
};


export default ButtonIcon;
