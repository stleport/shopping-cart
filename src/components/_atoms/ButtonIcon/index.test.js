import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ButtonIcon from '.';

function ButtonIconSetup(newProps = {}) {
  const props = {
    iconName: 'plus',
    iconAlign: 'right', 
    iconFloat: 'floated',
    iconColor: 'dark',
    enabled: true,
    bgColor: 'dark',
    onClick: jest.fn(),
    buttonStyle: '',
    onChangeQuantity: jest.fn()
  };

  const ButtonIconJSX = (
    <ButtonIcon {...{ ...props, ...newProps }} />
  );

  const ButtonIconComponent = mount(ButtonIconJSX);
  return {
    ButtonIconComponent,
    ButtonIconJSX,
    props
  };
}

describe('ButtonIcon', () => {
  it('renders correctly ButtonIcon component', () => {
    const { ButtonIconJSX } = ButtonIconSetup();
    const ButtonIconComponent = renderer.create(ButtonIconJSX).toJSON();
    expect(ButtonIconComponent).toMatchSnapshot();
  });
  it('onChangeQuantity handler should be call on plus button click', () => {
    const { ButtonIconComponent, props } = ButtonIconSetup();
    ButtonIconComponent.find('button').simulate('click');
    expect(props.onChangeQuantity).toHaveBeenCalled();
  });
});
