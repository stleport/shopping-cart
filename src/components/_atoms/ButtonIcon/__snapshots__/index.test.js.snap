// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon renders correctly ButtonIcon component 1`] = `
<button
  className=""
  onClick={[MockFunction]}
  type="button"
>
  <img
    alt="plus"
    className="st-Edit-button__icon--dark"
    fill="white"
    src="plus.svg"
  />
</button>
`;
